extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js

```diff
@@ -219,6 +220,51 @@ export class AmpAdXOriginIframeHandler {
       }
     });
 
+    if (this.element_.attributes.type.nodeValue == 'adsense' ||
```

Comment:
`this.element_.getAttribute`. Also, we already have some sort of "google reporter" instance going around. I believe I saw it in many places. I believe we just need to refactor this code into that class and call it from all three places.